spec/unit/resource_controller/data_access_spec.rb

```diff
@@ -225,5 +225,15 @@
     it "should assign nested attributes once" do
       expect(subject.posts.size).to eq(1)
     end
+
+    context "given authorization scope" do
+      let(:authorization) { controller.send(:active_admin_authorization) }
+
+      before { allow(authorization).to receive(:scope_collection) { |collection| collection.where(age: "42") } }
```

Comment:
What about using `expect` instead to ensure the method is called so if it's not then the test case fails? By using `allow` it means the method call is optional.
```suggestion
      before { expect(authorization).to receive(:scope_collection) { |collection| collection.where(age: "42") } }
```

Reply:
I have mixed feelings about using `expect` in `before` blocks.
Before block are for context, we should not "test" anything in them.
If we really need to be sure it is called from the scope collection I feel like it would require a proper dedicated test.

I guess that's the difference in approaches between "integration" (it's not really the case here but you get the point) test, rather than pure unit tests.

Here if we have a scope where your age is 42, then your output is a user with age 42. Little chance it came from everything else than the authorization scope. I tend to focus on what's in what's out vs what is called with which arguments.

It's a bit like line 214
```
let(:controller) do
  rc = Admin::UsersController.new
  allow(rc).to receive(:params) do
    params
  end
  rc
end
```

Open to talk about it

Reply:
I would have preferred that in the test case than a `before` block but no matter. To clarify, I reviewed the added lines only, as I'm not familiar with this file. In my opinion, using `allow` is a red flag because it can cause false positives. I'd rather a test fail loudly if a condition isn't meant which is a requirement for this test case. If that approach is used elsewhere and that's preferred though that's fine. I just point this out of caution.

Could you please rebase and push? Since this was opened we had updated our CI config, sorry.

Reply:
I don't have a strong opinion either, specially if using `allow` is already the most common style in this kind of tests. I also sympathize with the idea of making tests less dependent on the implementation, so that if you refactor the implementation the test still passes.

However I agree with @javierjulio that `expect` in this case is better. If it's not necessary to call this method here, and the call ends up being removed in the future to some refactor, keeping the `allow` will allow the test to still pass, but will most likely result in dead test code, since the line will be most likely left in there.

But no big deal, I'm happy with both versions.

Reply:
Then I guess the best of both worlds would be to remove it from the before block and put it in the testing block

Will update and rebase when I found some time thanks for your input guys ✌️